tools/delete_mri_upload.pl

```diff
@@ -0,0 +1,578 @@
+    exit $NeuroDB::ExitCodes::DB_SETTINGS_FAILURE;
+}
+
+#==================================#
+# Establish database connection    #
+#==================================#
+my $dbh = &NeuroDB::DBI::connect_to_db(@Settings::db);
+
+my $dataDirBasepath    = NeuroDB::DBI::getConfigSetting(\$dbh,'dataDirBasepath');
+$dataDirBasepath =~ s!/$!!;
+my $tarchiveLibraryDir = NeuroDB::DBI::getConfigSetting(\$dbh,'tarchiveLibraryDir');
+$tarchiveLibraryDir =~ s!/$!!;
+
+my $query = "SELECT m.TarchiveID, t.ArchiveLocation, m.SessionID "
+    .       "FROM mri_upload m "
+    .       "JOIN tarchive t USING (TarchiveID) "
+    .       "WHERE m.UploadID = ?";
+my $sth = $dbh->prepare($query);
+$sth->execute($uploadID);
+my($tarchiveID, $archiveLocation, $sessionID) = $sth->fetchrow_array();
+
+if($sth->rows == 0) {
+	print STDERR "No upload found in table mri_upload with upload ID '$uploadID'\n";
+	exit $NeuroDB::ExitCodes::INVALID_ARG;
+}
+
+if(!$tarchiveID) {
+	print STDERR "No tarchive ID found in table mri_upload for upload with ID $uploadID\n";
+	exit $NeuroDB::ExitCodes::INVALID_ARG;
+}
+
+if(!defined $tarchiveLibraryDir && $archiveLocation !~ /^\//) {
+    print STDERR "Cannot determine absolute path for archive '$archiveLocation' "
+        . "since config setting 'tarchiveLibraryDir' is not set. Aborting.\n";
+    exit $NeuroDB::ExitCodes::MISSING_CONFIG_SETTING;
+}
+
+$archiveLocation = "$tarchiveLibraryDir/$archiveLocation" unless $archiveLocation =~ /^\//;
+
+#===================================================#
+# Make sure there are no entries in files_qc_status #
+# and feedback_mri_comments for the files in the    #
+# archive                                           #
+#===================================================#
+
+if(&hasQcOrComment($dbh, $uploadID, $tarchiveID)) {
+	print STDERR "Cannot delete upload '$uploadID': there is QC information"
+        . " associated to that archive\n";
+	exit $NeuroDB::ExitCodes::INVALID_ARG;
+}
+
+#===========================================#
+# Backup all files found in                 #
+# tables files and files_intermediary for   #
+# the specified upload_id                   #
+#===========================================#
+
+my $filesRef             = &getFilesRef($dbh, $tarchiveID, $dataDirBasepath);
+my $intermediaryFilesRef = &getIntermediaryFilesRef($dbh, $tarchiveID, $dataDirBasepath);
+my $picFilesRef          = &getPicFilesRef($dbh, $tarchiveID, $dataDirBasepath);
+
+unless($noBackup) {
+    &backupFiles($uploadID, $archiveLocation, $filesRef, $intermediaryFilesRef, $picFilesRef);
+}
+&deleteUploadInDatabase($dbh, $uploadID, $tarchiveID, $sessionID, $intermediaryFilesRef, $picFilesRef);
+&deleteUploadFiles($archiveLocation, $filesRef, $intermediaryFilesRef, $picFilesRef);
+
+print "Upload $uploadID successfully deleted.\n";
+exit $NeuroDB::ExitCodes::SUCCESS;
+
+=pod
+
+=head3 hasQcOrComment($dbh, $tarchiveID)
+
+Determines if a tarchive has QC information associated to it by looking at the
```

Comment:
use either `C<tarchive>` or `DICOM archive`. 

Also, technically, it is not the tarchive that has the QC information but the MINC files issued from the DICOM archive. Would clarify that as it could be confusing.

Reply:
Ok